sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/SketchFrequencies.java

```diff
@@ -0,0 +1,544 @@
+     * and the greater the confidence, the greater the depth.
+     *
+     * @param epsilon the error relative to the total number of distinct elements
+     * @param confidence the confidence in the result to not exceed the relative error
+     */
+    public CountMinSketchFn<InputT> withAccuracy(double epsilon, double confidence) {
+      if (epsilon <= 0D) {
+        throw new IllegalArgumentException("The relative error must be positive");
+      }
+
+      if (confidence <= 0D || confidence >= 1D) {
+        throw new IllegalArgumentException("The confidence must be comprised between 0 and 1");
+      }
+      return new CountMinSketchFn<InputT>(this.inputCoder, epsilon, confidence);
+    }
+
+    @Override public Sketch<InputT> createAccumulator() {
+      return new Sketch<InputT>(this.epsilon, this.confidence);
+    }
+
+    @Override public Sketch<InputT> addInput(Sketch<InputT> accumulator, InputT element) {
+      accumulator.add(element, inputCoder);
+
+      return accumulator;
+    }
+
+    @Override public Sketch<InputT> mergeAccumulators(Iterable<Sketch<InputT>> accumulators) {
+      Iterator<Sketch<InputT>> it = accumulators.iterator();
+      Sketch<InputT> first = it.next();
+      CountMinSketch mergedSketches = first.sketch;
+      try {
+        while (it.hasNext()) {
+          mergedSketches = CountMinSketch.merge(mergedSketches, it.next().sketch);
+        }
+      } catch (FrequencyMergeException e) {
+        // Should never happen because every instantiated accumulator are of the same type.
+        throw new IllegalStateException("The accumulators cannot be merged !" + e.getMessage());
+      }
+      first.sketch = mergedSketches;
+
+      return first;
+    }
+
+    /** Output the whole structure so it can be queried, reused or stored easily. */
+    @Override public Sketch<InputT> extractOutput(Sketch<InputT> accumulator) {
+      return accumulator;
+    }
+
+
+    @Override public Coder<Sketch<InputT>> getAccumulatorCoder(CoderRegistry registry,
+                                                               Coder inputCoder) {
+      return new CountMinSketchCoder<InputT>();
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+      builder
+              .add(DisplayData.item("width", width)
+                      .withLabel("width of the Count-Min sketch array"))
+              .add(DisplayData.item("depth", depth)
+                      .withLabel("depth of the Count-Min sketch array"))
+              .add(DisplayData.item("eps", epsilon)
+                      .withLabel("relative error to the total number of elements"))
+              .add(DisplayData.item("conf", confidence)
+                      .withLabel("confidence in the relative error"));
+    }
+  }
+
+  /**
+   * Wrapper of StreamLib's Count-Min Sketch to fit with Beam requirements.
```

Comment:
Update comment:
`Wrap StreamLib's Count-Min Sketch to support counting all user types by hashing the encoded user type using the supplied deterministic coder. This is required since objects in Apache Beam are considered equal if their encodings are equal.`